### ISSUE TITLE
Move 401/403 HTTP error code mapping to restClient

### ISSUE THREAD

**Issue by yannpom** [marmelab/admin-on-rest#523]
Hello,

Both HTTP status code `401` and `403` are treated the same and `/login` is pushed:

```
// src/sideEffect/saga/auth.js
        if (error.status === 401 || error.status === 403) {
            yield put(push('/login'));
        }
```

In my API the semantic is different between both those code:

- `401` is used when the user is not authenticated.
- `403` is used when the user is authenticated but does not have the permission to access that ressource.

Having a `403` pushing the login page is annoying, and won't help as my user is already logged in. This can happen for exemple while fetching a relationship on an object and when this related object is not "readable" by the user.

I was wondering if I'm doing something wrong with those HTTP status codes or if maybe `403` could be removed from this `if`, or at least on option somewhere?

Cheers,
Yann.

**Comment by leesei** [comment]
I also suggest 403 should be handled separately.
Maybe a default of "Permission denied" notification is better than logging the user out.

**Comment by remi13131** [comment]
I am also convinced that it should behave this way. HTTP spec is pretty clear :
https://tools.ietf.org/html/rfc7235#section-3.1
> The 401 (Unauthorized) status code indicates that the request has not
   been applied because it lacks valid authentication credentials for
   the target resource. 
https://tools.ietf.org/html/rfc7231#section-6.5.3
> The 403 (Forbidden) status code indicates that the server understood
   the request but refuses to authorize it.  A server that wishes to
   make public why the request has been forbidden can describe that
   reason in the response payload (if any).

**Comment by fzaninotto** [comment]
the fact that different API use 401 and 403 status codes for different purposes makes me think that the logic mapping HTTP status codes with special actions in admin-on-rest (like logout) should be part of the `restClient`. I agree that this would be a welcome enhancement.

**Comment by remi13131** [comment]
Awesome thank you very much for this commit !

**Comment by fzaninotto** [comment]
Fixed by #542 (in the `next` branch, currently unreleased).
